File: 3676-smallest-number-with-all-set-bits/3676-smallest-number-with-all-set-bits.cpp
class Solution {
public:
    bool check(int n){
        int c = 0;
        int tc = 0;
        while(n != 0){
            if(n & 1){
                c += 1;
            }
            n = n >> 1;
            tc += 1;
        }
        if(c == tc) return true;
        else return false;
    }
    int smallestNumber(int n) {
        while(1){
            if(check(n)) return n;
            n += 1;
        }
    }
};

File: 3676-smallest-number-with-all-set-bits/3676-smallest-number-with-all-set-bits.java
class Solution {
    boolean setBits(int n){
        int nc = 0, c = 0;
        while(n != 0){
            if((n & 1) == 1){
                c += 1;
            }
            nc += 1;
            n = n >> 1;
        }
        if(nc == c) return true;
        return false;
    }
    public int smallestNumber(int n) {

        while(true){
            if(setBits(n) == true){
                return n;
            }
            n += 1;
        }
    }
}